src/battle_ai_main.c

```diff
@@ -1788,6 +1788,7 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
         case EFFECT_RESTORE_HP:
         case EFFECT_SOFTBOILED:
         case EFFECT_ROOST:
+        case EFFECT_JUNGLE_HEALING:
```

Comment:
Should get its own check. Fails only if both the attacker and partner's HP is full, as well as both of them not having a status condition.